### ISSUE TITLE
Support go routine scheduler locking or goroutine aware breakpoints

### ISSUE THREAD

**Issue by cucaroach** [go-delve/delve#2830]
In C++ its common to use gdb's set-scheduler-locking to make it so that only the current thread runs when you hit continue.    This is nice when there's lots of other threads hitting the same code that you aren't interested in.   Scheduler locking has pitfalls though, if the current goroutine needs to wait on another to finish it would just hang.   A better solution (that hopefully would be easier to implement) would be conditional goroutine aware breakpoints.   The idea is you would set a breakpoint A when the context you want to debug happens and B somewhere down stack and B would only fire if A was hit up the stack on the same goroutine.   An enhancement might be to have it when A's stack frame popped breakpoint B would no longer be active but that wouldn't be required for this to be useful.

**Comment by aarzilli** [comment]
Closing, duplicate of #1529 (it can not be done without assistence from the target process runtime).

**Comment by aarzilli** [comment]
PS. regarding:

>  A better solution (that hopefully would be easier to implement) would be conditional goroutine aware breakpoints

this already exists, you can set `runtime.curg.goid == whatever` as a condition on a breakpoint (but I guess we never documented it).

**Comment by aarzilli** [comment]
PPS. however conditional breakpoints will slow down the execution of the program if the code is executed a lot by many goroutines (potentially by a lot, it depends).
